src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpRemoveUnnecessaryImportsService.Rewriter.cs

```diff
@@ -83,13 +84,29 @@ public override SyntaxNode DefaultVisit(SyntaxNode node)
                                     // want to preserve.
                                     currentUsings[nextIndex] = nextUsing.WithLeadingTrivia(leadingTrivia);
                                 }
+
+                                passedLeadngTrivia = true;
                             }
                             else
                             {
                                 finalTrivia = leadingTrivia;
                             }
                         }
                     }
+                    else
+                    {
+                        if (onlyDeletedUsingsBefore)
+                        {
+                            if (i > 0 && !passedLeadngTrivia)
```

Comment:
Can you comment the logic and what this is doing to be clear what the expectations are here. I'm struggling to work out what `passedLeadingTrivia` is for, without digging in to `ShouldPreserveTrivia` and what the rest of this whole class is doing. When does passedLeadingTrivia become true?